pipelines/base/enterprise-contract.yaml

```diff
@@ -0,0 +1,65 @@
+# The purpose of this pipeline is to execute the verify-enterprise-contract-v2 task for container
+# images that are built but not automatically released in order to provide early feedback to users.
+# When auto release is enabled, the task is executed by the release pipeline immediately after the
+# container images are built, thus it is not necessary to execute the task via this pipeline.
+apiVersion: tekton.dev/v1beta1
+kind: Pipeline
+metadata:
+  name: enterprise-contract
+  labels:
+    "skip-hacbs-test": "true"
+spec:
+  params:
+    - name: SNAPSHOT
+      type: string
+      description: |
+        Spec section of an ApplicationSnapshot resource. Not all fields of the
+        resource are required. A minimal example:
+          {
+            "components": [
+              {
+                "containerImage": "quay.io/example/repo:latest"
+              }
+            ]
+          }
+        Each "containerImage" in the "components" array is validated.
+    - name: POLICY_CONFIGURATION
+      type: string
+      description: |
+        Name of the policy configuration (EnterpriseContractConfiguration
+        object) to use. `namespace/name` or `name` syntax supported. If
+        namespace is omitted the namespace where the task runs is used.
+      default: ec-policy
+    - name: SSL_CERT_DIR
+      type: string
+      description: |
+        Path to a directory containing SSL certs to be used when communicating
+        with external services. This is useful when using the integrated registry
+        and a local instance of Rekor on a development cluster which may use
+        certificates issued by a not-commonly trusted root CA. In such cases,
+        "/var/run/secrets/kubernetes.io/serviceaccount" is a good value. Multiple
+        paths can be provided by using the ":" separator.
+      default: ""
+  results:
+    - name: REPORT
+      value: "$(tasks.verify.results.REPORT)"
+  tasks:
+    - name: verify
+      params:
+        - name: POLICY_CONFIGURATION
+          value: "$(params.POLICY_CONFIGURATION)"
+        - name: IMAGES
+          value: "$(params.SNAPSHOT)"
+        - name: SSL_CERT_DIR
+          value: "$(params.SSL_CERT_DIR)"
+        # Tekton only creates results for successful tasks. Since the goal of this pipeline
+        # is not to gate an operation, but to provide feedback to users, do not fail if there
+        # are violations. This ensures users can view the full validation report.
+        - name: STRICT
+          value: "false"
+        # Public Key should come from the Policy Configuration
+        - name: PUBLIC_KEY
+          value: ""
+      taskRef:
+        # Does this get automatically linked to the bundle? If not how do other pipelines work?
```

Comment:
In the general case referring to the task by name shouldn't work if the task definition isn't installed first. Our approach for example integration test pipelines was to refer the task by bundle image since we can't assume that the correct task definition would be present.

@Michkov do you know if there's any replacement happening here or does the task definition need to be present on the cluster?

Reply:
My apologies, I just realized that this is a leftover comment. The replacement happens when running the script `hack/build-and-push.sh`. 

When I run this locally, this is what it turns into:
```
$ jq '.spec.tasks[].taskRef'
{
  "bundle": "quay.io/lucarval/appstudio-tasks:2022-08-30-102651-3",
  "name": "verify-enterprise-contract-v2"
}
```

So if you run the Pipeline from the pipeline bundle that is automatically generated for every commit to main in this repo, it should be all set. Here's how I run this Pipeline:
```
apiVersion: tekton.dev/v1beta1
kind: PipelineRun
metadata:
  generateName: enterprise-contract-
spec:
  params:
  - name: SNAPSHOT
    value: |
      {
        "components": [
          {"containerImage": "quay.io/lucarval/tekton-test:ede69244d7"}
        ]
      }
  - name: SSL_CERT_DIR
    value: /var/run/secrets/kubernetes.io/serviceaccount
  pipelineRef:
    bundle: quay.io/lucarval/build-templates-bundle:2022-08-30-102651@sha256:b0dee279a793ee1d4f3842ec33f7154c3bfaf21674cdbcc23eaed72547156d3b
    name: enterprise-contract
```

You don't have to know which image to use in taskRef, but you do need to know the bundle in which the pipeline definition lives. Does that make sense?

Reply:
Yes, that makes sense - the bundle with the pipeline is what needs to be set in the `IntegrationTestScenario` CR. Do you know if this will be a set image with a tag?

Reply:
The repo will always be `quay.io/redhat-appstudio/build-templates-bundle`. There isn't a "latest" tag. Images are tagged based on git commit ID. For instance, right now, the latest image is `quay.io/redhat-appstudio/build-templates-bundle:510fa6e99f1fa1f816c96354bbaf1ad155c6d9c3`

This is something that would have to be constantly updated. There's some gitops workflows triggered by this repo that creates pull requests in other repos when new images are available. Maybe it's something we can look into for this case, too.

Reply:
Could we tag it latest at the quay.io level using the quay api post bundle push and just keeping moving the latest tag on every bundle push?

Reply:
There has been push back with such approach as it makes things nondeterministic, which outweighs its convenience. I would say that a "latest" tag would be good for trying things out, but I would be skeptic about using it in any official workflow.

@Michkov, are you opposed to the idea of also tagging the image in `quay.io/redhat-appstudio/build-templates-bundle` with a latest tag?

Reply:
Maybe we can use different tag eg. `dev`

Reply:
Yeah, I second adding a tag, otherwise [DEVHAS-161](https://issues.redhat.com/browse/DEVHAS-161) might get complicated to implement.

Reply:
Ack. Let's aim for that then. I'd like to do that as a separate PR if possible.

Reply:
Agreed, we can start testing the current form by using the sha. Thanks for all the inputs everybody.